community/kernel/src/main/java/org/neo4j/graphdb/GraphDatabaseService.java

```diff
@@ -183,10 +187,55 @@
      * space).
      *
      * @return all relationship types in the underlying store
-     * @deprecated this operation can be found in {@link GlobalGraphOperations} instead.
      */
-    @Deprecated
-    Iterable<RelationshipType> getRelationshipTypes();
+    ResourceIterable<RelationshipType> getRelationshipTypes();
+
+    /**
+     * Returns all relationship types currently in use in the underlying store. Relationship types are
+     * added to the underlying store the first time they are used in a successfully committed
+     * {@link Node#createRelationshipTo node.createRelationshipTo(...)}. Note that this method is
+     * guaranteed to return all known relationship types, and it guarantees that it won't return
+     * "historic" relationship types that no longer have any relationships in the graph.
+     *
+     * @return all relationship types in use in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUse();
```

Comment:
Exactly. The other "get all" methods from GGO are called `getAll....`, so for consistency that would make sense. However in this case that would break the API, but that's fine since `getRelationshipTypes()` happens to be deprecated anyway (although for a different reason).

We shouldn't have methods exposing unused stuff, should we? The fact that it's the current behaviour is quite strange. We don't have any other method returning unused nodes/relationships/properties.